utils/include/CRequestHandler.cpp

```diff
@@ -0,0 +1,154 @@
+    try
+    {
+
+        if (!TryValidateRequest(request_str))
+        {
+            response.TryFormateResponse(answer, "",
+                EFrameError::INCORRECT_REQUEST); 
+            //log and return true
+	    	CLOG_ERROR("Invalid request from the client!!!");
+            return false;
+        }
+        EFrameError error_from_frame = GetErrorCodeFromFrame(request_str);
+        if (EFrameError::NONE != error_from_frame)
+        {
+            response.TryFormateResponse(answer, "",
+                error_from_frame);
+            //log and return true
+            CLOG_ERROR("Invalid request from the client!!!");
+            return false;
+        }
+
+        switch (AnalyzeRequestType(request))
+        {
+        case ERequestType::ALL_DATA:
+            if (!ExecuteRequest(answer, std::make_shared<CRequestExcAllData>(
+                request_str, std::make_shared<CDataProvider>(m_data_base))))
+            {
+                response.TryFormateResponse(answer, "",
+                    EFrameError::INCORRECT_REQUEST); 
+                return false;
+            }
+            CLOG_DEBUG("All data request from the client");
+
+            break;
+        case ERequestType::DISKS_DATA:
+            if(!ExecuteRequest(answer, std::make_shared<CRequestExcDiskData>(
+                request_str, std::make_shared<CDataProvider>(m_data_base))))
+            {
+                response.TryFormateResponse(answer, "",
+                    EFrameError::INCORRECT_REQUEST); 
+                return false;
+            }
+            CLOG_DEBUG("Disks data request from the client");
+
+            break;
+        case ERequestType::PROCESSES_DATA:
+            if(!ExecuteRequest(answer, std::make_shared<CRequestExcProcessData>(
+                request_str, std::make_shared<CDataProvider>(m_data_base))))
+            {
+                response.TryFormateResponse(answer, "",
+                    EFrameError::INCORRECT_REQUEST); 
+                return false;
+            }
+            CLOG_DEBUG("Processes data request from the client");
+
+            break;
+        default:
+        {
+            response.TryFormateResponse(answer, "", 
+                EFrameError::LOST_REQUEST);
+            CLOG_ERROR("Default case, incorrect error");
+            return false;
+        }
+        }
+    }
+    // todo: kind of exceptio
+    catch (...)
+    {
+        return response.TryFormateResponse(answer, answer, 
+            EFrameError::INCORRECT_REQUEST);
+    }
+    CLOG_DEBUG_END_FUNCTION();
+    // could be a problem
+    return response.TryFormateResponse(answer, answer, EFrameError::NONE);
+}
+
+EFrameError CRequestHandler::GetErrorCodeFromFrame(
+    const std::string& request_str)
+{
+    nlohmann::json request = nlohmann::json::parse(request_str);
+    int err = request[GlobalVariable::c_frame_error].get<int>();
+    return EFrameError(err);
+}
+
+bool CRequestHandler::TryValidateRequest(const std::string& request_str)
```

Comment:
Should be reworked

Reply:
Added more sense to this method: now it's not only checking key's name, but also check is frame store something beside keys' names that request shouldn't store